mnd/mnd/registry/patients/mnd_admin_forms.py

```diff
@@ -158,13 +158,10 @@ def save(self, commit=True):
         if instance and instance.pk:
             db_instance = PrimaryCarer.objects.get(pk=instance.pk)
             email_changed = db_instance.email != email
-            existing_user = CustomUser.objects.filter(username=db_instance.email, is_active=True).exists()
-            carer_instance_update = email_changed and existing_user
-            if carer_instance_update:
+            if email_changed:
                 # get or create a new primary carer entry if email is changed
-                # and for that email address the carer registered already
-                new_carer = PrimaryCarer.objects.filter(email__iexact=email.lower()).first()
-                self.instance.pk = new_carer.pk if new_carer else None
+                existing_carer = PrimaryCarer.objects.filter(email__iexact=email.lower()).first()
```

Comment:
If the patient changes the email and the PrimaryCarer doesn't exist, then when a patient changes the email to another that doesn't have an associated carer then instance is `None`:

```
    def all_forms_valid(self, forms):
        ret_val = super().all_forms_valid(forms)
        formset_keys = [self.PATIENT_INSURANCE_KEY, self.PRIMARY_CARER_KEY, self.PREFERRED_CONTACT_KEY]
        for key in formset_keys:
            instance = forms[key].save(commit=False)
>           instance.patient = self.object
            instance.save()
            if key == self.PRIMARY_CARER_KEY:
                self._handle_primary_carer_relationship(forms[key], instance)

        return ret_val
```

Reply:
Fixed the issue.